pkg/notifications/email.go

```diff
@@ -0,0 +1,74 @@
+// Copyright 2024 The Sigstore Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package notifications
+
+import (
+	"context"
+
+	"github.com/sigstore/rekor-monitor/pkg/identity"
+	"github.com/wneessen/go-mail"
+)
+
+// EmailNotificationInput extends the NotificationPlatform interface to support
+// found identity notification by sending emails to a specified user.
+type EmailNotificationInput struct {
+	RecipientEmailAddress string
+	SenderEmailAddress    string
+	SenderSMTPUsername    string
+	SenderSMTPPassword    string
+	SMTPHostURL           string
```

Comment:
Keeping SMTP username/password as fields since those seem pretty necessary in most use cases so feels nice to have outside of custom options, which should be used generally much more sparingly